test/test_linalg.py

```diff
@@ -2972,6 +2972,20 @@ def run_test(shape):
         for batch, (m, n) in itertools.product(batches, product(ns, ns)):
             run_test((*batch, m, n))
 
+    @skipCUDAIfNoCusolver  # MAGMA backend doesn't work in this case
+    @skipCUDAIfRocm
+    @skipCPUIfNoLapack
+    @dtypes(torch.float32, torch.float64, torch.complex64, torch.complex128)
+    def test_svd_memory_allocation(self, device, dtype):
+        # test for https://github.com/pytorch/pytorch/issues/61949
+        # the problem was that tensors of incorrect size were allocated and then narrowed
+        m = 3
+        n = 2**20
+        a = make_tensor((m, n), dtype=dtype, device=device)
+        # the following should run without errors
+        result = torch.linalg.svdvals(a)
+        result = torch.linalg.svd(a, full_matrices=False)
```

Comment:
Let's also check the `out=` variant

Reply:
Added the `out=` variants and it also revealed that `some` argument was not correct for `svdvals_out` (https://github.com/pytorch/pytorch/pull/62022#discussion_r676879529).